app/modules/individuals/parameters.py

```diff
@@ -76,5 +181,29 @@ def replace(cls, obj, field, value, state):
         elif field == 'featuredAssetGuid' and util.is_valid_guid(value):
             obj.set_featured_asset_guid(UUID(value, version=4))
             ret_val = True
+        elif field == 'names':
+            from app.modules.names.models import Name
+
+            if (
+                not isinstance(value, dict)
+                or 'guid' not in value.keys()
+                or not util.is_valid_guid(value['guid'])
+                or ('context' not in value.keys() and 'value' not in value.keys())
+            ):
+                abort(
+                    code=HTTPStatus.UNPROCESSABLE_ENTITY,
+                    message='value must contain keys "guid" and at least one of: "context", "value"',
+                )
+            name = Name.query.get(value['guid'])
```

Comment:
This name extraction and validation is in this file 3 times, this would be easier to read if all three functions called one helper function